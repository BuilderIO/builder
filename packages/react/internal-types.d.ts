declare module 'rollup-plugin-replace';
declare module 'rollup-plugin-node-resolve';
declare module 'rollup-plugin-re';
declare module 'rollup-plugin-alias';

declare module 'lib/on-change.js';
declare module '../../lib/on-change';
