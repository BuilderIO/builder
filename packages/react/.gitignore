node_modules
coverage
.nyc_output
.DS_Store
*.log
.vscode
.idea
dist
compiled
.awcache
.rpt2_cache

docs/.nojekyll
docs/classes
docs/modules.md
docs/modules
docs/README.md
docs/interfaces/BuilderComponentState.md