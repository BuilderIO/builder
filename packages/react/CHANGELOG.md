### 1.1.50

- update `@builder.io/sdk` to `1.1.26` (adds bugfix to `noTrack` correctly ignoring session cookies)
