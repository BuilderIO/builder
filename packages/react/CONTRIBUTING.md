We're really glad you're reading this, because we need volunteer developers to help this project come to fruition. üëè

Refer to [the v2 contribution guide](../sdks/docs/DEVELOP.md) for more up-to-date information

## Instructions

These steps will guide you through contributing to this project:

- Fork the repo
- Clone it and install dependencies

      		git clone https://github.com/YOUR-USERNAME/builder-react
      		yarn install

Keep in mind that after running `yarn install` the git repo is reset. So a good way to cope with this is to have a copy of the folder to push the changes, and the other to try them.

Make and commit your changes. Make sure the commands npm run build and npm run test:prod are working.

Finally send a [GitHub Pull Request](https://github.com/builderio/builder-react/compare?expand=1) with a clear list of what you've done (read more [about pull requests](https://help.github.com/articles/about-pull-requests/)). Make sure all of your commits are atomic (one feature per commit).

Refer to 