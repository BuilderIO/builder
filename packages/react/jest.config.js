module.exports = {
  testEnvironment: 'jest-environment-jsdom',
  setupFilesAfterEnv: ['@testing-library/jest-dom/extend-expect'],
  testMatch: ['**/?(*.)test.ts?(x)'],
  globals: {
    'ts-jest': {
      tsConfig: 'tsconfig.json',
      diagnostics: false,
    },
  },
  testEnvironment: 'node',
  preset: 'ts-jest',
  coveragePathIgnorePatterns: ['/node_modules/', '/test/'],
  coverageThreshold: {
    global: {
      branches: 25,
      functions: 25,
      lines: 25,
      statements: 25,
    },
  },
  collectCoverage: true,
};
