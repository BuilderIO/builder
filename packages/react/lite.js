/**
 * To use the lite build:
 *
 *    // Change all imports from '@builder.io/react' to '@builder.io/react/lite'
 *    import { BuilderComponent } from '@builder.io/react/lite';
 *
 *    // Import only what built-in components you like
 *    import '@builder.io/react/dist/lib/src/blocks/Button';
 *    import '@builder.io/react/dist/lib/src/blocks/Columns';
 *
 *
 * You should generally use this in conjunction with `customInsertMenu` moe, e.g.:
 *    Builder.register('editor.settings', { customInsertMenu: true })
 * to ensure that no components are displayed that aren't imported
 *
 *
 * E.g. github.com/BuilderIO/builder/blob/master/examples/react-design-system/src/builder-settings.js#L9:L9
 *
 */
module.exports = require('./dist/builder-react-lite.cjs.js');
