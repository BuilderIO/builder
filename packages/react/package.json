{
  "name": "@builder.io/react",
  "version": "1.1.32-alpha.0",
  "description": "",
  "keywords": [],
  "main": "dist/builder-react.cjs.js",
  "unpkg": "dist/builder-react.unpkg.js",
  "module": "dist/builder-react.es5.js",
  "typings": "dist/types/src/builder-react.d.ts",
  "author": "Steve Sewell <steve@builder.io>",
  "repository": {
    "type": "git",
    "url": "https://github.com/BuilderIO/builder"
  },
  "license": "MIT",
  "engines": {
    "node": ">=6.0.0"
  },
  "tags": [
    "react",
    "builder.io"
  ],
  "scripts": {
    "_unlink": "npm unlink *",
    "_install": "npm install",
    "lint": "tslint -t codeFrame '../**/*.ts' 'test/**/*.ts'",
    "prebuild": "rimraf dist",
    "build": "NODE_ENV=production tsc --module commonjs && rollup -c rollup.config.ts",
    "build:docs": "typedoc --out docs --target es6 --theme minimal --mode file src",
    "start": "rollup -c rollup.config.ts -w",
    "test": "jest",
    "test:watch": "jest --watch",
    "test:prod": "npm run lint && npm run test -- --coverage --no-cache",
    "deploy-docs": "ts-node tools/gh-pages-publish",
    "report-coverage": "cat ./coverage/lcov.infoâ‰ˆ | coveralls",
    "commit": "git-cz",
    "release": "npm run build && ALLOW_PUBLISH=true npm publish",
    "release:patch": "npm run build && npm version patch && ALLOW_PUBLISH=true npm publish",
    "release:nightly": "npm run build && npm version prerelease --no-git-tag-version && ALLOW_PUBLISH=true npm publish --tag nightly",
    "release:dev": "npm run build && npm version prerelease --no-git-tag-version && ALLOW_PUBLISH=true npm publish --tag dev",
    "semantic-release": "semantic-release",
    "semantic-release-prepare": "ts-node tools/semantic-release-prepare",
    "precommit": "lint-staged"
  },
  "lint-staged": {
    "{src,test}/**/*.ts": [
      "prettier --write",
      "git add"
    ]
  },
  "config": {
    "commitizen": {
      "path": "node_modules/cz-conventional-changelog"
    },
    "validate-commit-msg": {
      "types": "conventional-commit-types",
      "helpMessage": "Use \"npm run commit\" instead, we use conventional-changelog format :) (https://github.com/commitizen/cz-cli)"
    }
  },
  "jest": {
    "transform": {
      ".(ts|tsx|jsx)": "<rootDir>/node_modules/ts-jest/preprocessor.js"
    },
    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(ts|tsx)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js"
    ],
    "testURL": "http://localhost/",
    "coveragePathIgnorePatterns": [
      "/node_modules/",
      "/test/"
    ],
    "moduleDirectories": [
      "<rootDir>/node_modules"
    ],
    "coverageThreshold": {
      "global": {
        "branches": 25,
        "functions": 25,
        "lines": 25,
        "statements": 25
      }
    },
    "collectCoverage": true
  },
  "devDependencies": {
    "@builder.io/block-publish": "^1.1.2",
    "@testing-library/jest-dom": "^4.0.0",
    "@testing-library/react": "^9.1.0",
    "@types/hash-sum": "^1.0.0",
    "@types/jest": "^22.2.3",
    "@types/node": "^10.17.13",
    "@types/object-hash": "^1.3.0",
    "@types/react": "^16.4.1",
    "@types/react-dom": "^16.0.7",
    "@types/react-test-renderer": "^16.9.2",
    "@types/rollup-plugin-commonjs": "^9.2.0",
    "@types/rollup-plugin-json": "^3.0.1",
    "@types/rollup-plugin-node-resolve": "^4.1.0",
    "@types/rollup-plugin-sourcemaps": "^0.4.0",
    "colors": "^1.1.2",
    "commitizen": "^2.9.6",
    "coveralls": "^3.0.0",
    "cross-env": "^5.0.1",
    "cz-conventional-changelog": "^2.0.0",
    "husky": "^0.14.0",
    "jest": "^22.0.2",
    "lint-staged": "^7.0.0",
    "prettier": "^1.4.4",
    "prompt": "^1.0.0",
    "react-test-renderer": "^16.13.1",
    "replace-in-file": "^3.0.0-beta.2",
    "rimraf": "^2.6.1",
    "rollup": "^1.32.1",
    "rollup-plugin-alias": "^1.5.1",
    "rollup-plugin-babel-minify": "^9.1.1",
    "rollup-plugin-commonjs": "^9.0.0",
    "rollup-plugin-json": "^3.0.0",
    "rollup-plugin-node-resolve": "^3.0.0",
    "rollup-plugin-re": "^1.0.7",
    "rollup-plugin-replace": "^2.0.0",
    "rollup-plugin-sourcemaps": "^0.4.2",
    "rollup-plugin-terser": "^5.2.0",
    "rollup-plugin-typescript2": "^0.11.1",
    "rollup-plugin-uglify": "^6.0.4",
    "semantic-release": "^17.2.3",
    "ts-jest": "^22.4.6",
    "ts-node": "^6.0.0",
    "tslint": "^5.8.0",
    "tslint-config-prettier": "^1.1.0",
    "tslint-config-standard": "^7.0.0",
    "typedoc": "^0.11.0",
    "typescript": "^3.8.3",
    "validate-commit-msg": "^2.12.2"
  },
  "//": "TODO: maybe make @builder.io/sdk peerdependency",
  "optionalDependencies": {
    "create-react-context": "^0.2.3",
    "node-fetch": "^2.3.0",
    "prop-types": "^15.7.2"
  },
  "dependencies": {
    "@builder.io/sdk": "^1.1.21-alpha.0",
    "@emotion/core": "^10.0.17",
    "hash-sum": "^2.0.0",
    "preact": "^10.1.0",
    "react": ">=14",
    "react-dom": ">=14",
    "tslib": "^1.10.0",
    "vm2": "^3.6.4"
  },
  "gitHead": "4d96fbc32864698afbb355ab991c6d90be991951"
}
