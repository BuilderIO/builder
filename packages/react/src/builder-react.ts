import './scripts/init-editing';

import { builder, Builder } from '@builder.io/sdk';
export { BuilderElement } from '@builder.io/sdk';

Builder.isReact = true;

export { BuilderBlocks } from './components/builder-blocks.component';
export { BuilderBlock as BuilderBlockComponent } from './components/builder-block.component';
export { BuilderContent } from './components/builder-content.component';
import { BuilderComponent, onChange } from './components/builder-component.component';
export { BuilderStoreContext, BuilderStore } from './store/builder-store';
export { BuilderMetaContext } from './store/builder-meta';
export { BuilderAsyncRequestsContext } from './store/builder-async-requests';
export { BuilderBlock } from './decorators/builder-block.decorator';

export * from './functions/update-metadata';

export { withBuilder } from './functions/with-builder';
export { withChildren } from './functions/with-children';
export { noWrap } from './functions/no-wrap';

export { BuilderComponent as BuilderPage, onChange };
export { BuilderComponent };
export { BuilderComponent as RenderContent };

export { Text } from './blocks/Text';
export { Slot as Dropzone } from './blocks/Slot';
export { Fragment } from './blocks/Fragment';
export { Columns } from './blocks/Columns';
export { Embed } from './blocks/Embed';
export { CustomCode } from './blocks/CustomCode';
export { Image, getSrcSet } from './blocks/Image';
export { Video } from './blocks/Video';
export { Symbol } from './blocks/Symbol';
export { Button } from './blocks/Button';
export { Section } from './blocks/Section';
export { StateProvider } from './blocks/StateProvider';
export { Router } from './blocks/Router';
export { Mutation } from './blocks/Mutation';

export { Form } from './blocks/forms/Form';
export { FormInput } from './blocks/forms/Input';
export { FormSubmitButton } from './blocks/forms/Button';
export { Label } from './blocks/forms/Label'; // advanced?
export { FormSelect } from './blocks/forms/Select'; // advanced?
export { TextArea } from './blocks/forms/TextArea';
export { Img } from './blocks/raw/Img';
export { RawText } from './blocks/raw/RawText';

export { stringToFunction } from './functions/string-to-function';

export { builder, Builder };
export default builder;
