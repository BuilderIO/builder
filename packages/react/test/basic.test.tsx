import * as React from 'react'
import { renderToString } from 'react-dom/server'
import { render } from '@testing-library/react'
import { Builder, builder } from '@builder.io/sdk'
import { BuilderPage } from '../src/builder-react'
import { el, block } from './functions/render-block'

builder.init('null')

import '@testing-library/jest-dom/extend-expect'

describe('Dummy test', () => {
  it('tests run correctly', () => {
    expect(true).toBeTruthy()
  })
})

const server = (cb: () => void) => {
  Builder.isServer = true
  try {
    cb()
  } finally {
    Builder.isServer = false
  }
}

describe('Renders tons of components', () => {
  const blocks = [
    block('Columns', {
      columns: [{ blocks: [el()] }, { blocks: [el()] }]
    }),
    block('CustomCode', {
      code: '<!-- hello -->'
    }),
    block('Embed', {
      content: '<!-- hello -->'
    }),
    block('Symbol'),
    block('Router'),
    block('Image', { image: 'foobar' }),
    block('Form:Form'),
    block('Video', { video: 'foobar' }),
    block('Button', { text: 'foobar' }),
    block('Section', null, { children: [el()] }),
    block('Form:SubmitButton', { text: 'foobar' }),
    block('Form:Input', { type: 'text' }),
    block('Form:Label'),
    block('Form:Select'),
    block('Form:TextArea', { placeholder: 'foobar' }),
    block('Raw:Img', { image: 'foobar' })
  ]

  const getRenderExampleElement = () => (
    <BuilderPage
      model="page"
      content={{
        data: {
          blocks: blocks
        }
      }}
    />
  )

  it('works with dom', () => {
    const testApi = render(getRenderExampleElement())
  })
  it('works with SSR', () => {
    renderToString(getRenderExampleElement())
  })
})

describe('Data rendering', () => {
  const TEXT_STRING = 'Hello 1234'
  const bindingBlock = el({
    bindings: {
      'component.options.text': 'state.foo'
    },
    component: {
      name: 'Text'
    }
  })

  const getBindingExampleElement = () => (
    <BuilderPage
      model="page"
      data={{ foo: TEXT_STRING }}
      content={{
        data: {
          blocks: [bindingBlock]
        }
      }}
    />
  )

  it('works with dom', () => {
    const testApi = render(getBindingExampleElement())
    expect(testApi.getByText(TEXT_STRING)).toBeInTheDocument()
  })

  it('works with SSR', () => {
    server(() => {
      const renderedString = renderToString(getBindingExampleElement())
      expect(renderedString).toContain(TEXT_STRING)
    })
  })
})

describe('Content changes when new content provided', () => {
  const textA = 'textA'
  const textB = 'textB'
  const idA = 'id-a'
  const idB = 'id-b'

  it('Handles content passed and changed correctly', () => {
    const testApi = render(
      <BuilderPage
        model="page"
        content={{
          id: idA,
          data: {
            blocks: [block('Text', { text: textA })]
          }
        }}
      />
    )

    expect(testApi.getByText(textA)).toBeInTheDocument()

    testApi.rerender(
      <BuilderPage
        model="page"
        content={{
          id: idB,
          data: {
            blocks: [block('Text', { text: textB })]
          }
        }}
      />
    )
    expect(testApi.getByText(textB)).toBeInTheDocument()
  })

  it('Should be in controlled mode for null or underined content', () => {
    const testApi = render(<BuilderPage model="page" content={undefined} />)

    expect(testApi.queryByText(textB)).toBeNull()

    testApi.rerender(
      <BuilderPage
        model="page"
        content={{
          id: idB,
          data: {
            blocks: [block('Text', { text: textB })]
          }
        }}
      />
    )
    expect(testApi.getByText(textB)).toBeInTheDocument()
  })

  it('Should be in controlled mode for null or underined content', () => {
    const testApi = render(<BuilderPage model="page" />)

    expect(testApi.queryByText(textB)).toBeNull()

    testApi.rerender(
      <BuilderPage
        model="page"
        content={{
          id: idB,
          data: {
            blocks: [block('Text', { text: textB })]
          }
        }}
      />
    )
    expect(testApi.getByText(textB)).toBeInTheDocument()
  })
})
