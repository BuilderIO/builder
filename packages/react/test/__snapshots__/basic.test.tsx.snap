// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Builder Pixel Should NOT be added again if already present in blocks array 1`] = `
<div
  className="builder-component builder-component-id"
  data-name="page"
  data-source="Rendered by Builder.io"
  onClick={[Function]}
>
  <div
    builder-content-id="id"
    builder-model="page"
    className="builder-content"
    onClick={[Function]}
  >
    <div
      data-builder-component="page"
      data-builder-content-id="id"
    >
      
      <div
        builder-type="blocks"
        className="builder-blocks css-h47494"
        onClick={[Function]}
      >
        <img
          aria-hidden="true"
          builder-id="builder-pixel-4fzzzxjylrx"
          className="builder-block builder-pixel-4fzzzxjylrx css-1mvsfya"
          role="presentation"
          src="https://cdn.builder.io/api/v1/pixel?apiKey=null"
          style={Object {}}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Builder Pixel Should NOT be added if blocks array is empty 1`] = `
<div
  className="builder-component builder-component-id"
  data-name="page"
  data-source="Rendered by Builder.io"
  onClick={[Function]}
>
  <div
    builder-content-id="id"
    builder-model="page"
    className="builder-content"
    onClick={[Function]}
  >
    <div
      data-builder-component="page"
      data-builder-content-id="id"
    >
      
      <div
        builder-type="blocks"
        className="builder-blocks no-blocks css-h47494"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`Builder Pixel Should be added if pixel is missing and blocks array has other block(s) 1`] = `
<div
  className="builder-component builder-component-id"
  data-name="page"
  data-source="Rendered by Builder.io"
  onClick={[Function]}
>
  <div
    builder-content-id="id"
    builder-model="page"
    className="builder-content"
    onClick={[Function]}
  >
    <div
      data-builder-component="page"
      data-builder-content-id="id"
    >
      
      <div
        builder-type="blocks"
        className="builder-blocks css-h47494"
        onClick={[Function]}
      >
        <div
          builder-id="builder-270035a08d734ae88ea177daff3595c0"
          className="builder-block builder-270035a08d734ae88ea177daff3595c0 builder-has-component css-hgfgng"
          style={Object {}}
        >
          <span
            className="builder-text css-1qggkls"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p>some text...</p>",
              }
            }
            onBlur={[Function]}
            onFocus={[Function]}
            onInput={[Function]}
            onKeyDown={[Function]}
          />
        </div>
        <img
          aria-hidden="true"
          builder-id="builder-pixel-4fzzzxjylrx"
          className="builder-block builder-pixel-4fzzzxjylrx css-1mvsfya"
          role="presentation"
          src="https://cdn.builder.io/api/v1/pixel?apiKey=null"
          style={Object {}}
        />
      </div>
    </div>
  </div>
</div>
`;
